### PR TITLE
ast: add set default role

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Support `SET DEFAULT ROLE` to set default active roles for user.

### What is changed and how it works?
Add `SetDefaultRoleStmt` for `ast` which contain information for `executor` to do next step. 

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

### CODE REVIEW

File: ast/misc.go [hunk 1]
```diff
@@ -813,6 +814,58 @@ func (n *SetRoleStmt) Accept(v Visitor) (Node, bool) {
...
+
+func (n *SetDefaultRoleStmt) Restore(ctx *RestoreCtx) error {
+	ctx.WriteKeyWord("SET DEFAULT ROLE")
+	switch n.SetRoleOpt {
+	case SetRoleNone:
+		ctx.WriteKeyWord(" NONE")
+	case SetRoleAll:
+		ctx.WriteKeyWord(" ALL")
+	default:
+		ctx.WriteKeyWord("")
```
Comment (kennytm) [inline ast/misc.go]:
```suggestion
```

No need to write nothing 🙂

**Comment by jackysp** [comment]
Please address the comment, @imtbkcat .

File: ast/misc.go (hunk 1 above)
Comment (imtbkcat) [reply ast/misc.go]: @kennytm fixed.

**Comment by imtbkcat** [comment]
@kennytm this pr need an approve 😊

**Comment by kennytm** [comment]
Please resolve the merge conflict 🙂

**Comment by imtbkcat** [comment]
@kennytm conflict has been reslove, but approve was dismissed. Please approve again and merge this pr.

**Pull request merged**